drivers/zio/ft5336/ft5336.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2019, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zio.h>
+#include <init.h>
+#include <i2c.h>
+#include <logging/log.h>
+#include <drivers/zio/ft5336.h>
+
+LOG_MODULE_REGISTER(ft5336, CONFIG_ZIO_LOG_LEVEL);
+
+#define MAX_TOUCHES		(5)
+
+struct ft5336_config {
+	char *i2c_name;
+	u8_t i2c_address;
+};
+
+struct ft5336_data {
+	struct device *i2c;
+	struct k_work work;
+	struct device *dev;
+	ZIO_FIFO_BUF_DECLARE(fifo, struct ft5336_datum, CONFIG_FT5336_FIFO_SIZE);
+};
+
+struct ft5336_raw {
+	u8_t gest_id;
+	u8_t td_status;
+	struct {
+		u8_t xh;
+		u8_t xl;
+		u8_t yh;
+		u8_t yl;
+		u8_t reserved[2];
+	} touch[MAX_TOUCHES];
+} __packed;
```

Comment:
How do we know which of the 5 possible touch point events are valid? Is it worth having some sort of `.valid` flag to indicate that a particular record is valid, or should we be checking for 0 values to know if `touch[x]` contains useful data (one finger used, three fingers, etc.)?